jlm/llvm/opt/alias-analyses/RegionAwareModRefSummarizer.cpp

```diff
@@ -55,199 +89,110 @@ class RegionAwareModRefSummarizer::Statistics final : public util::Statistics
+    GetTimer(NonReentrantAllocaSetsTimer_).stop();
   }
 
-  CallSummary(const CallSummary &) = delete;
-
-  CallSummary(CallSummary &&) = delete;
-
-  CallSummary &
-  operator=(const CallSummary &) = delete;
-
-  CallSummary &
-  operator=(CallSummary &&) = delete;
-
-  [[nodiscard]] const rvsdg::SimpleNode &
-  GetCallNode() const noexcept
+  void
+  StartCreateExternalModRefSet()
   {
-    return *CallNode_;
+    AddTimer(CreateExternalModRefSetTimer_).start();
   }
 
   void
-  AddMemoryNodes(const util::HashSet<const PointsToGraph::MemoryNode *> & memoryNodes)
+  StopCreateExternalModRefSet()
   {
-    MemoryNodes_.UnionWith(memoryNodes);
+    GetTimer(CreateExternalModRefSetTimer_).stop();
   }
 
-  const util::HashSet<const PointsToGraph::MemoryNode *> &
-  GetMemoryNodes() const noexcept
+  void
+  StartAnnotationStatistics()
   {
-    return MemoryNodes_;
+    AddTimer(AnnotationTimer_).start();
   }
 
   void
-  SetPossiblyRecursive() noexcept
+  StopAnnotationStatistics()
   {
-    PossiblyRecursive_ = true;
+    GetTimer(AnnotationTimer_).stop();
   }
 
-  bool
-  IsPossiblyRecursive() const noexcept
+  void
+  StartSolvingStatistics()
   {
-    return PossiblyRecursive_;
+    AddTimer(SolvingTimer_).start();
   }
 
-  /**
-   * @return the SCC index the function containing this call belongs to in the call graph
-   */
-  size_t
-  GetCallGraphSccIndex() const noexcept
+  void
+  StopSolvingStatistics()
   {
-    return CallGraphSccIndex_;
+    GetTimer(SolvingTimer_).stop();
   }
 
-  /**
-   * Creates a new CallSummary representing the given call node.
-   * @param callNode the call node represented by the summary
-   * @param sccIndex the index of the SCC the function containing the call belongs to, in the module
-   * call graph
-   * @return the newly created RegionSummary
-   */
-  static std::unique_ptr<CallSummary>
-  Create(const rvsdg::SimpleNode & callNode, size_t sccIndex)
+  static std::unique_ptr<Statistics>
+  Create(const rvsdg::RvsdgModule & rvsdgModule, const PointsToGraph & pointsToGraph)
   {
-    return std::make_unique<CallSummary>(callNode, sccIndex);
+    return std::make_unique<Statistics>(rvsdgModule, pointsToGraph);
   }
-
-private:
-  // The call node represented by this summary
-  const rvsdg::SimpleNode * CallNode_;
-
-  // The set of memory locations that may be read from or written to by this call
-  util::HashSet<const PointsToGraph::MemoryNode *> MemoryNodes_;
-
-  // Which SCC in the call graph does the function containing this call belong to
-  size_t CallGraphSccIndex_;
-  // If it is possible for this call to be recursive, this flag will be set to true
-  bool PossiblyRecursive_;
 };
 
 /**
- * Class containing a summary of memory locations possibly read from or written to inside a given
- * region
+ * Class representing the set of MemoryNodes that may be modified or referenced by some operation,
+ * or within some region.
  */
-class RegionSummary final
+class ModRefSet final
```

Comment:
Not sure whether this class is necessary. I do not really the value of it. It only wraps util::HashSet

Reply:
I guess it is useful to have this explicit abstraction maybe.